src/ScreenManager.cpp

```diff
@@ -262,23 +262,25 @@ ScreenManager::ScreenManager()
 
 ScreenManager::~ScreenManager()
 {
-	LOG->Trace( "ScreenManager::~ScreenManager()" );
-	LOG->UnmapLog( "ScreenManager::TopScreen" );
+	LOG->Trace("ScreenManager::~ScreenManager()");
+	LOG->UnmapLog("ScreenManager::TopScreen");
 
-	SAFE_DELETE( g_pSharedBGA );
-	for( unsigned i=0; i<g_ScreenStack.size(); i++ )
+	SAFE_DELETE(g_pSharedBGA);
+	for (LoadedScreen& screen : g_ScreenStack)
```

Comment:
It does not like `for (const LoadedScreen& screen : g_ScreenStack)`. Some of these loops are modifying the things it iterates though I believe

```
Error	C3490	'm_pScreen' cannot be modified because it is being accessed through a const object	ITGmania	ScreenManager.cpp	272		

Error (active)	E0137	expression must be a modifiable lvalue	ITGmania	ScreenManager.cpp	272		
```

Reply:
Sounds good. Yeah I wasn't sure what some of the macros were doing so the ones that can't be const is fine to drop that qualifier.